### ISSUE TITLE
FormControls initialized multiple times

### ISSUE THREAD

**Issue by juristr** [formly-js/ngx-formly#801]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[x] support request
```

**Current behavior**

I added a custom form control (wrapping ng-select) into my formly config on my app. Whenever a binding is set like

```
theModel = {
   cityId: 345 
}
```

..in the `writeValue(..)` function (of the `ControlValueAccessor`) I take the `cityId` and fetch the according city instance from the backend or somewhere else. That's when I noticed that my `writeValue(..)` is actually being called like 30-40 times in my huge form, always with the value of the data-bound `cityId`.

**Expected behavior**

It should actually just be called once. Unless I'm doing something wrong here in the formly binding.

**Minimal reproduction of the problem with instructions**
I built a simple custom form control not doing much actually, just for the purpose of intercepting the calls in the `writeValue(..)`.

https://stackblitz.com/edit/angular-ngxformly-repeat-s8cm39?file=app%2Fapp.component.ts

If the the `app.compnent.ts`'s Formly config we just add

```
{
   key: 'name',
   type: 'custom'
},
```

..then it gets called just once, actually twice: the first time with a null value and then with the proper one.

If instead, we just add

```
{
      fieldGroup: [
         {
             key: 'name',
             type: 'custom'
        }
   ]
},
```

..basically wrapping it in a `fieldGroup`, then it already gets called 3 times, 1st time again with a null value, and then twice with the data-bound value.

It gets even more strange if we add another arbitrary form control, like

```
{
  fieldGroup: [
    {
      key: 'name',
      type: 'custom'
    }
  ]
},
{
  fieldGroup: [
    {
      key: 'firstname',
      type: 'input'
    }
  ]
}
```

..it gets called 4 times, 1 with null, 3 with data. And that continues as we add more formly bindings.

> As a side-note: I'm using `fieldGroup` in my setup, to add the bootstrap `row` class grouping together multiple form controls on the same row.

* **Angular version:** 5.1.0

**Comment by aitboudad** [comment]
good catch, fixed in `2.0.0-rc.41`.

**Comment by juristr** [comment]
Ur a machine 😄 . I've seen ur response, just forgot to write back. I think I'll try to incorporate this change into our codebase today. thx man 👍
